src/components/createsecret.vue

```diff
@@ -0,0 +1,313 @@
+      <div class="mt-4">
+        <input
+          class="mb-2 appearance-none border rounded py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
+          type="text"
+          required
+          placeholder="Secret name"
+          v-model="secretName"
+        />
+      </div>
+      <p
+        v-if="formSubmitted && !isSecretNameValid"
+        class="text-xs text-red-700 mb-2"
+      >
+        {{
+          keyPairsErrors[0].secretName || 'Please enter a valid secret name.'
+        }}
+      </p>
+      <p
+        v-if="formSubmitted && isDuplicateSecretName"
+        class="text-xs text-red-700 mb-2"
+      >
+        Secret name is already in use.
+      </p>
+      <div v-for="(pair, index) in secretvalues" :key="index">
+        <div class="flex items-center mt-4">
+          <input
+            class="appearance-none border rounded py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
+            type="text"
+            required
+            placeholder="Key"
+            v-model="pair.key"
+          />
+          <input
+            class="ml-2 appearance-none border rounded py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
+            type="text"
+            required
+            placeholder="Value"
+            v-model="pair.value"
+          />
+          <button
+            class="btn-gray font-bold py-1 px-3 ml-2 rounded"
+            v-if="secretvalues.length > 1"
+            @click="removeKeyValuePair(index)"
+            :disabled="secretvalues.length === 1"
+          >
+            Remove
+          </button>
+        </div>
+        <div v-if="formSubmitted">
+          <p
+            class="text-xs text-red-700 mb-2"
+            v-if="formSubmitted && keyPairsErrors[index].keyValuePair"
+          >
+            {{ keyPairsErrors[index].keyValuePair }}
+          </p>
+          <p
+            class="text-xs text-red-700 mb-2"
+            v-if="formSubmitted && keyPairsErrors[index].key"
+          >
+            {{ keyPairsErrors[index].key }}
+          </p>
+        </div>
+      </div>
+      <div class="flex mt-4">
+        <button
+          class="bg-green-600 hover:bg-green-700 text-white font-bold py-2 px-4 rounded"
+          @click="addKeyValuePair"
+        >
+          Add Key-Value Pair
+        </button>
+        <button
+          class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 ml-2 rounded"
+          type="submit"
+        >
+          Save
+        </button>
+      </div>
+    </form>
+  </div>
+</template>
+<script lang="ts">
+import { ApiError, SecretResponse, useAPI } from '../app/api';
+import { projectGroupSettingsLink, projectSettingsLink } from '../util/link';
+import { PropType, Ref, computed, onMounted, ref, toRefs } from 'vue';
+import { useRouter } from 'vue-router';
+import { useAsyncState } from '@vueuse/core';
+import { useAppState } from '../app/appstate';
+import _ from 'lodash';
+
+interface SecretValues {
```

Comment:
is this just a Record<string, string>?

Reply:
> is this just a Record<string, string>?

For me its easy using an array to getting data from the form.